test/jdk/java/net/httpclient/http3/H3StreamLimitReachedTest.java

```diff
@@ -23,27 +23,63 @@
 
 /*
  * @test id=with-default-wait
+ * @summary this test verifies that the correct connection is
+ *          used when a request is retried on a new connection
+ *          due to stream limit reached.
+ *          The maxBidiStreams limit is artificially set to 1.
+ *          This configuration uses the default wait for a stream
+ *          to become available before retrying.
+ *          Different configurations are tested if possible,
+ *          with both HTTP/3 only and HTTP/2 HTTP/3 with altsvc.
+ *          In one case the HTTP/3 only server will be listening
+ *          on the same port as the HTTP/2 server, with the
+ *          HTTP/2 server advertising an AltService on a different
+ *          port. In another case the AltService will be on the
+ *          same port as the HTTP/2 server and the HTTP/3 server
+ *          will be on a different port. In all case, the test
+ *          verifies that the right connection is picked up
+ *          for the retry.
+ * @bug 8372951
+ * @comment this test also tests bug 8372951
```

Comment:
Nit - i think we can remove these `@comment` lines because the `@bug` already states that this tests `8372951`. If we do retain this comment line, then it would mean we have to keep updating it (along with `@bug` every time there a new bug fix this test verifies).